bin/common.bash

```diff
@@ -334,14 +335,28 @@ validate_version() {
     exit 1
   fi
   ck8s_version=$(yq '.global.ck8sVersion' "${merged_config}")
+  cluster_version=$(get_apps_version "${1}" 2>/dev/null || true)
   if [[ -z "$ck8s_version" ]]; then
     log_error "ERROR: No version set. Run init to generate config."
     exit 1
   elif [ "${ck8s_version}" != "any" ] &&
     [ "${version}" != "${ck8s_version}" ]; then
-    log_error "ERROR: Version mismatch. Run init to update config."
-    log_error "Config version: ${ck8s_version}"
-    log_error "Welkin Apps version: ${version}"
+    log_error "ERROR: Version mismatch. Run upgrade to update config."
+    log_error "Welkin Apps cluster version: ${cluster_version}"
+    log_error "Welkin Apps config version: ${ck8s_version}"
+    log_error "Welkin Apps repository version: ${version}"
+    exit 1
+  fi
+  if [[ -z "${cluster_version}" ]]; then
+    log_warning "Welkin Apps cluster version:     Unknown"
+    log_warning "Welkin Apps config version:      ${ck8s_version}"
+    log_warning "Welkin Apps repository version:  ${version}"
+  elif [[ "${cluster_version}" != "${version%.*}" ]]; then
+    # TODO double check whether patch version is included here
```

Comment:
```suggestion
    # TODO double check whether patch version is included here
```
Is this still needed?

Reply:
Would be good to check an extra time if all the `$version_*` variables include the patch version or not, so it's consistent.